### PR TITLE
Update Taxon PaperClip attributes on attachment destroy

### PR DESCRIPTION
**Pull request by RyanofWoods**
## Summary
When using Paperclip, calling destroy on the attachment removes it from
from the storage location, such as S3, and leaves dirty changes on the
Paperclip attributes, such as icon_file_name.

If a store used `.present?` on the icon, it would return continue to
return `true`, as it looks at the file_name attribute [1], yet the
icon no longer exists on storage (`.exists? => false`).

A view such as the backend Taxon edit view, could therefore attempt to
show the icon, but it would fail and show a broken icon image.

Note:
- an assertion was not added for `.present?` as ActiveStorage's API
  works a little differently;
- stores using Paperclip may have Taxons with Paperclip attributes
  present, yet the image on storage no longer exists. For these Taxons, a broken image and the remove button will appear in the backend. Frontend could be affected as well.

[1] https://github.com/kreeti/kt-paperclip/blob/91923a6ee1bd692696b51ea839f2f2578cb485b4/lib/paperclip/attachment.rb#L357-L361

### Screenshots
| Description | Screenshot |
|--|--|
| Attached icon before deletion | Screenshot 2023-05-24 at 08 36 19|
| Attached icon after deletion but before fix | Screenshot 2023-05-24 at 08 37 55 |
| Attempting to remove a broken Taxon's icon which does not exist | Screenshot 2023-05-24 at 09 58 46 |
| Attached icon after deletion and after fix | Screenshot 2023-05-24 at 09 59 14 |

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: core/spec/models/spree/taxon_spec.rb
Comment (kennyadsl) [inline core/spec/models/spree/taxon_spec.rb:26]: Should we remove this line?

File: core/lib/generators/solidus/install/install_generator.rb
Comment (kennyadsl) [inline core/lib/generators/solidus/install/install_generator.rb:99]: ❤️

**Review by kennyadsl — APPROVED** [review]
Thanks @RyanofWoods!!! This makes perfect sense, and we will probably backport this bug fix in all the supported Solidus versions.

I left a question/comment for a line in the spec that might not be necessary anymore, but let me know if that's correct or I missed something. 

🙏

File: core/spec/models/spree/taxon_spec.rb
Comment (RyanofWoods) [reply core/spec/models/spree/taxon_spec.rb]: Definitely 🤦 😁 Thanks!

**Comment by RyanofWoods** [comment]
One more note, a Taxon's PaperClip icon attachment will now trigger some Taxon callbacks due to the `save`:
https://github.com/solidusio/solidus/blob/a8f56d585dd50f802827df53630e4a08e10bcfe2/core/app/models/spree/taxon.rb#L27

A bit unfortunate, but I don't think it will be an issue.

**Comment by kennyadsl** [comment]
> but I don't think it will be an issue.

I don't think either. On the other side, it will invalidate the cache which might be a good thing when the icon is removed.

**Comment by RyanofWoods** [comment]
Good question @waiting-for-dev
1. I am not really sure why paperclip is not saving the associated record. Note, that the documented `save` is actually for the attachment though. Also, it seems that the maintainers don't really know why these attributes are not being set to `nil` 🤔 https://github.com/kreeti/kt-paperclip/pull/91#discussion_r1027060962. Let's hold off on merging this, while I  investigate further 
2. Good idea, I will do that now 👍

**Comment by RyanofWoods** [comment]
A few findings @waiting-for-dev. Note `exists?` checks the actual storage, while `.present?` checks the attachment's filename attribute

Dec 6, 2019:
Replaced taxons/_form attachment `.exists?` to `.present?`
https://github.com/solidusio/solidus/commit/484f7d58f731060ae2a08c069ac99f7e205bed38

Jan 6, 2021:
Replaced taxons/_form attachment `.present?` to `.exists?`
https://github.com/solidusio/solidus/commit/6e19cd795db4779088275bc2965455917f8d29e9

Because of the latter, I think it's important to continue using `.present?`. We therefore need  the taxon's `icon_file_name` to be saved to `nil`, so the storage is in sync with attribute, and `present?` is correct.

From the [kt-paperclip](https://github.com/kreeti/kt-paperclip) documentation, it seems that they intentionally do not call save on the taxon instance because they want the attachment to act like an "attribute".
```
The intent behind it was to keep setup as easy as possible and to treat files as much like other attributes as possible. This means they aren't saved to their final locations on disk, nor are they deleted if set to nil, until ActiveRecord::Base#save is called.
```

**Review by waiting-for-dev — APPROVED** [review]
Thanks for your investigation, @RyanofWoods!

**Comment by waiting-for-dev** [comment]
Oh, backport failed probably because the last commit in this PR had already been backported :disappointed: We'll need to do it manually...

**Comment by RyanofWoods** [comment]
Sorry @waiting-for-dev 😢 I removed the commit, but I see that I pushed to my fork instead of `nebulab` 🙁

**Comment by waiting-for-dev** [comment]
No worries, @RyanofWoods, my fault for not checking before. Luckily, it has easy solution :slightly_smiling_face:

**Pull request merged**
